### PR TITLE
Feature/settings

### PR DESCRIPTION
**Pull request by jungerj**
# New settings / better configuration
The following settings are _new_:

- direction: set motor direction
- encoder position: specify whether the encoder is on the joint or motor side of the gears.
- An explanation of some config parameters that are not completely self explanatory has been added.

The following other things have been changed / added:

- Sanity check for configuration object. Includes an intuitive printout in either green or red such that people actually use this feature (instead of only checking a return value which is often omitted in certain projects)
- A small fix for the mode of operations when 'use_multiple_modes_of_operation' is set to true.
- Better example config file

### CODE REVIEW

File: src/elmo_ethercat_sdk/Configuration.cpp [hunk 1]
```diff
@@ -19,11 +19,46 @@
  */
 
 #include <iomanip>
+#include <string>
 
 #include "elmo_ethercat_sdk/Configuration.hpp"
 
 namespace elmo {
 
+bool Configuration::sanityCheck(bool silent) {
```
Comment (firesurfer) [inline src/elmo_ethercat_sdk/Configuration.cpp:28]: Thats a nice feature

File: src/elmo_ethercat_sdk/Configuration.cpp [hunk 2]
```diff
@@ -19,11 +19,46 @@
...
+#define CHECK_AND_INFORM(bool_condition)                                       \
```
Comment (firesurfer) [inline src/elmo_ethercat_sdk/Configuration.cpp:29]: I think you could use a function (or lambda function for this)

**Comment by firesurfer** [comment]
See my coments. Looks fine otherwise

File: src/elmo_ethercat_sdk/Configuration.cpp (hunk 2 above)
Comment (jungerj) [reply src/elmo_ethercat_sdk/Configuration.cpp:29]: Yes. The string that is printed would however have to be specified manually for each condition in that case.
I think the macro here is OK as it is available only locally and '#undef'-ed at the and of the function.

**Pull request merged**